iphone/Classes/TiCalendarRecurrenceRule.m

```diff
@@ -53,8 +53,12 @@ - (id)valueForUndefinedKey:(NSString *)key
     return currRule.calendarIdentifier;
   } else if ([key isEqualToString:@"end"]) {
     EKRecurrenceEnd *end = currRule.recurrenceEnd;
-    NSDictionary *recurrenceEnd = [NSDictionary dictionaryWithObjectsAndKeys:[TiUtils UTCDateForDate:end.endDate], @"endDate",
-                                                NUMUINTEGER(end.occurrenceCount), @"occurrenceCount", nil];
+    NSDictionary *recurrenceEnd = nil;
+    if (end.occurrenceCount > 0) {
+      recurrenceEnd = [NSDictionary dictionaryWithObjectsAndKeys:NUMUINTEGER(end.occurrenceCount), @"occurrenceCount", nil];
+    } else if (end.endDate != nil) {
+      recurrenceEnd = [NSDictionary dictionaryWithObjectsAndKeys:[TiUtils UTCDateForDate:end.endDate], @"endDate", nil];
+    }
```

Comment:
@kopiro Hey Flavio, quick question on this one: What happens if the `EKRecurrenceEnd` has both `occurrenceCount` and `endDate`? Won't the `endDate` get lost with this change? Maybe add an `if`-statement to check for both to retain the old behavior as well, adding the partial constructions as well?